scripts/core/eventQueryKernelTimestamps.yml

```diff
@@ -0,0 +1,103 @@
+type: struct
+desc: "Event query kernel timestamps results properties"
+version: "1.6"
+name: $x_event_query_kernel_timestamps_results_ext_properties_t
+base: $x_base_properties_t
+members:
+    - type: $x_kernel_timestamp_result_t*
+      name: pTimestampsBuffer
+      desc: "[in,out][optional][range(0, *pCount)] pointer to destination buffer"
+      init: nullptr
+    - type: uint64_t*
+      name: pSynchronizedTimestampsBuffer
+      desc: "[in,out][optional][range(0, *pCount)] pointer to destination buffer"
+      init: nullptr
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Query an event's timestamp value on the host, with domain preference."
+version: "1.6"
+class: $xEvent
+name: QueryKernelTimestampsExt
+details:
+    - "For collecting *only* kernel timestamps, the application must ensure the event was created from an event pool that was created using $X_EVENT_POOL_FLAG_KERNEL_TIMESTAMP flag."
+    - "For collecting synchronized timestamps, the application must ensure the event was created from an event pool that was created using $X_EVENT_POOL_FLAG_KERNEL_MAPPED_TIMESTAMP flag. Kernel timestamps are also available from this type of event pool, but there is a performance cost."
+    - "The destination memory will be unmodified if the event has not been signaled."
+    - "The application may call this function from simultaneous threads."
+    - "The implementation of this function must be thread-safe."
+    - "The implementation of this function should be lock-free."
+    - "The implementation must support $X_extension_event_query_kernel_timestamps."
+    - "The implementation must return all timestamps for the specified event and device pair."
+    - "The implementation must return all timestamps for all sub-devices when device handle is parent device."
+    - "The implementation may return all timestamps for sub-devices when device handle is sub-device or may return 0 for count."
+params:
+    - type: $x_event_handle_t
+      name: hEvent
+      desc: "[in] handle of the event"
+    - type: $x_device_handle_t
+      name: hDevice
+      desc: "[in] handle of the device to query"
+    - type: uint32_t*
+      name: pCount
+      desc: |
+            [in,out] pointer to the number of event packets available.
+                - This value is implementation specific.
+                - if `*pCount` is zero, then the driver shall update the value with the total number of event packets available.
+                - if `*pCount` is greater than the number of event packets available, the driver shall update the value with the correct value.
+                - Buffer(s) for query results must be sized by the application to accommodate a minimum of `*pCount` elements.
+    - type: $x_event_query_kernel_timestamps_results_ext_properties_t*
+      name: pResults
+      desc: |
+            [in][optional] pointer to event query properties structure(s).
```

Comment:
@joshuaranjan why is this out? it's a props struct

Reply:
@aravindksg 
I was thinking since resultsProps.pSynchronizedTimestampsBuffer is updated by the driver, when this API is called, may be this should be [out].